src/main/java/bridge/controller/BridgeController.java

```diff
@@ -0,0 +1,65 @@
+package bridge.controller;
+
+import bridge.service.BridgeRandomNumberGenerator;
+import bridge.domain.BridgeMove;
+import bridge.domain.BridgeSize;
+import bridge.domain.GameCommand;
+import bridge.dto.GameResultDto;
+import bridge.dto.RoundResultDto;
+import bridge.service.BridgeGame;
+import bridge.view.InputView;
+import bridge.view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BridgeController {
+
+    private final BridgeGame bridgeGame;
+    private final OutputView outputView;
+    private final InputView inputView;
+
+    public BridgeController(BridgeGame bridgeGame, OutputView outputView, InputView inputView) {
+        this.bridgeGame = bridgeGame;
+        this.outputView = outputView;
+        this.inputView = inputView;
+    }
+
+    public void run() {
+        outputView.printGameStart();
+
+        BridgeSize bridgeSize = inputView.readBridgeSize();
+        bridgeGame.makeBridge(bridgeSize, new BridgeRandomNumberGenerator());
+
+        List<RoundResultDto> map = new ArrayList<>();
+        GameResultDto gameResultDto = gamePlay(bridgeSize, map);
+        outputView.printResult(gameResultDto);
+    }
+
+    private GameResultDto gamePlay(BridgeSize bridgeSize, List<RoundResultDto> map) {
+        int totalTryCount = 1;
+        int roundCount = 1;
+        while (true) {
+            BridgeMove bridgeMove = inputView.readMoving();
+            RoundResultDto move = bridgeGame.move(bridgeMove, roundCount);
+            map.add(move);
```

Comment:
```While(true)```로 반복하고 종료 조건을 if로 처리하도록 설계되어있네

개인적으로 이러한 방식보다는 다음과 같은 방식이 가독성이 좋다고 생각함

```suggestion
        while (isGameEnd || gameCommand.equals(GameCommand.QUIT)) {

        }
```

반복문이 뭘 위해서 돌고 있는지, 언제 종료되는지 이렇게 표시하면 조금 더 명확할 듯 한데 어떻게 생각하뉘

나도 이게 무조건 맞다라고 생각이 드는 건 아니라서 너 생각도 궁금하네 🤔

Reply:
제 나쁜 습관 중에 하나를 정확하게 보신 것 같습니다😅
반복의 상황에서 일단 무한 루프를 돌린뒤 조건에 맞게 break 하는 방식을 먼저 고려하는데 작성하신 코드처럼
`반복문이 뭘 위해서 돌고 있는지, 언제 종료되는지`를 정확하게 알고 있어야 가독성도 좋고, 유지보수도 좋아질 것 같습니다😁